doc/doxygen/headers/reordering.h

```diff
@@ -19,17 +19,68 @@
  * @brief A module describing how deal.II consistently orients Triangulation
  * objects.
  *
- * <h3>Statement of problems</h3>
- *
- * Triangulations in deal.II have a special structure, in that there are not
- * only cells, but also faces, and in 3d also edges, that are objects of their
- * own right. Faces and edges have unique orientations, and they have a
- * specified orientation also with respect to the cells that are adjacent.
- * Thus, a line that separates two cells in two space dimensions does not only
- * have a direction, but it must also have a well-defined orientation with
- * respect to the other lines bounding the two quadrilaterals adjacent to the
- * first line. Likewise definitions hold for three dimensional cells and the
- * objects (lines, quads) that separate them.
+ * @warning The implementation of orientation should be considered an internal
+ * detail of the library. Normal users should not need to use the features
+ * describd in this module: instead, classes like QProjector use orientation
+ * information to consistently compute values on faces and lines.
+ *
+ * <h2>Orientation of Lines</h2>
+ *
+ * A Triangulation is built not just of cells but also of lower-dimensional
+ * objects. In particular, a line in 2D is a face which may be shared by two
+ * cells, whereas in 3D a line may be shared by an arbitrary number of cells.
+ *
+ * Lines (i.e., faces in 2D) are implicitly defined by the vertex numbering of
+ * the cell on which they are defined (i.e., there is no separate line or face
+ * object). This information is encoded in various ReferenceCell functions. For
+ * example, the first face of a triangle with nodes {0, 1, 2} is {0, 1}.
+ * Similarly, the first face of a triangle with nodes {1, 0, 3} is {1, 0}.
+ * By itself, this would be inconsistent, since the same line would be defined
+ * in two diferent ways. To solve this problem deal.II also stores an
+ * <tt>unsigned char</tt> whose value encodes the orientation of the line. In
+ * this particular case the orientation of the {1, 0} line is
+ * ReferenceCell::default_combined_face_orientation() whereas the orientation of
+ * the {0, 1} line is ReferenceCell::reversed_combined_line_orientation().
+ *
+ * <h2>Orientation of Faces</h2>
+ *
+ * In deal.II, we express the orientation of an object with three booleans:
+ * orientation, rotate, and flip. The default values for these are true, false,
+ * and false. These are represented either as individual booleans or as a
+ * combined orientation packed into a single <tt>unsigned char</tt>: the
+ * combined format places orientation in the least significant bit, then rotate,
+ * then flip.
```

Comment:
I don't think you should specify the bits here. Just refer to the pack and unpack functions from bools to the unsigned char.